src/test/messenger/xio_client.cc

```diff
@@ -114,11 +114,14 @@ int main(int argc, const char **argv)
 	else
 	  dstrategy = new QueueStrategy(2);
 
+        Messenger::Proplist opts;
+        opts.insert(Messenger::Property("xio_portal_threads", 1));
+        opts.insert(Messenger::Property("xio_max_conns_per_portal", 2));
```

Comment:
I know this is just an example, but I hope that for `ceph_osd.cc` we can avoid using magic numbers. So the actual values should come from ceph.conf; how do you imagine that working?

You mentioned the distinction between 'light' vs. 'workhorse' messengers, so would it be a set of variables like `xio_{light,heavy}_portal_threads` and `xio_{light,heavy}_max_conns_per_portal`?

If this is going to be the strategy, maybe all we really need to pass into `Messenger::create()` is this light/heavy flag. Then we could avoid building up these property lists for each messenger, and the flags would be much more likely to be useful for other messengers.